packages/wallet-frontend/src/ui/forms/Input.tsx

```diff
@@ -1,29 +1,56 @@
-import { forwardRef, useId } from 'react'
+import { type ReactNode, forwardRef, useId } from 'react'
 import type { ComponentPropsWithoutRef } from 'react'
-
 import { Label } from './Label'
 import { FieldError } from './FieldError'
+import { cx } from 'class-variance-authority'
 
-type InputProps = Omit<ComponentPropsWithoutRef<'input'>, 'className'> & {
+type InputProps = ComponentPropsWithoutRef<'input'> & {
   label?: string
   error?: string
+  addOn?: ReactNode
+  trailing?: ReactNode
+  labelHint?: ReactNode
```

Comment:
I've been thinking of adding a `trailingIcon` or `isLoading` property. My though process:
- there is some async work that needs to be done for a certain input - the `isLoading` case
- we might want to add a trailing icon (error badge, show/hide for passwords) - the `trailingIcon` case

My main concern would be if we have trailing content and a trailing icon. Should we only allow  one if the trailing properties (ie: if `trailing` is present, do not accept `trailingIcon` and the other way around)?

I have been considering the possibility of implementing a `trailingIcon` or `isLoading` property. My reasoning behind this is as follows:
- there may be certain inputs that require asynchronous work - the `isLoading` case
- it may be useful to include a trailing icon (such as an error badge or a show/hide button for password) - the `trailingIcon` case

My primary concern is whether or not we should allow both trailing content and a trailing icon. Should we only permit one of these trailing options (e.g., if `trailing` is present, we do not accept `trailingIcon` and vice versa)?